### PR TITLE
esp_task_wdt_feed() did not reset fed_watchdog flag

### PR DESCRIPTION
**Pull request by honza-klu**
esp_task_wdt_feed() did not reset fed_watchdog flag properly after feeding real watchdog. It skipped last task in the list. Issue is fixed now.
I think that this error can be related to issue #906

### CODE REVIEW

**Comment by negativekelvin** [comment]
I think you meant to only remove `->next` and leave `!=`

File: components/esp32/task_wdt.c [hunk 1]
```diff
@@ -131,7 +131,7 @@ void esp_task_wdt_feed() {
         TIMERG0.wdt_feed=1;
         TIMERG0.wdt_wprotect=0;
         //Reset fed_watchdog status
-        for (wdttask=wdt_task_list; wdttask->next!=NULL; wdttask=wdttask->next) wdttask->fed_watchdog=false;
+        for (wdttask=wdt_task_list; wdttask=NULL; wdttask=wdttask->next) wdttask->fed_watchdog=false;
     }
```
Comment (PerMalmberg) [inline components/esp32/task_wdt.c]: Surely `wdttask=NULL` should be  `wdttask != NULL`

File: components/esp32/task_wdt.c [hunk 2]
```diff
@@ -131,7 +131,7 @@ void esp_task_wdt_feed() {
         TIMERG0.wdt_feed=1;
         TIMERG0.wdt_wprotect=0;
         //Reset fed_watchdog status
-        for (wdttask=wdt_task_list; wdttask->next!=NULL; wdttask=wdttask->next) wdttask->fed_watchdog=false;
+        for (wdttask=wdt_task_list; wdttask=NULL; wdttask=wdttask->next) wdttask->fed_watchdog=false;
```
Comment (PerMalmberg) [inline components/esp32/task_wdt.c]: Surely `wdttask=NULL` should be  `wdttask != NULL`

**Comment by honza-klu** [comment]
Yes, I did, It is fixed now.

**Comment by Spritetm** [comment]
Looks good, but could you please squash your two commits into one?

**Comment by Spritetm** [comment]
Oops, a colleague just pointed out we have some rework on the WDT code going on that also fixes this. WIll keep this open until that is merged, but we may not need this fix. Still thanks for pointing it out, though!

**Pull request closed without merging**